In [ ]:
!pip install sentence_transformers

In [ ]:
from sentence_transformers.cross_encoder import CrossEncoders

/usr/local/lib/python3.10/dist-packages/sentence_transformers/cross_encoder/CrossEncoder.py:11: TqdmExperimentalWarning: Using `tqdm.autonotebook.tqdm` in notebook mode. Use `tqdm.tqdm` instead to force console mode (e.g. in jupyter console)
  from tqdm.autonotebook import tqdm, trange


ImportError: cannot import name 'CrossEncoders' from 'sentence_transformers.cross_encoder' (/usr/local/lib/python3.10/dist-packages/sentence_transformers/cross_encoder/__init__.py)

In [ ]:
!pip install --upgrade sentence-transformers  # Upgrade to the latest version

from sentence_transformers import CrossEncoder

In [ ]:
!pip install peft

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 296.4/296.4 kB 5.6 MB/s eta 0:00:00


In [ ]:
!pip install --quiet transformers
from transformers import AutoTokenizer, AutoModelForSequenceClassification
import torch

model_name = "vectara/hallucination_evaluation_model"
tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
model = AutoModelForSequenceClassification.from_pretrained(model_name, trust_remote_code=True)

You are using a model of type HHEMv2Config to instantiate a model of type HHEMv2. This is not supported for all configurations of models and can yield errors.


ValueError: Unrecognized configuration class <class 'transformers_modules.vectara.hallucination_evaluation_model.2c44150de684dbcbc6eb2abd9e92b428c5fb3361.configuration_hhem_v2.HHEMv2Config'> to build an AutoTokenizer.
Model type should be one of AlbertConfig, AlignConfig, BarkConfig, BartConfig, BertConfig, BertGenerationConfig, BigBirdConfig, BigBirdPegasusConfig, BioGptConfig, BlenderbotConfig, BlenderbotSmallConfig, BlipConfig, Blip2Config, BloomConfig, BridgeTowerConfig, BrosConfig, CamembertConfig, CanineConfig, ChineseCLIPConfig, ClapConfig, CLIPConfig, CLIPSegConfig, ClvpConfig, LlamaConfig, CodeGenConfig, CohereConfig, ConvBertConfig, CpmAntConfig, CTRLConfig, Data2VecAudioConfig, Data2VecTextConfig, DbrxConfig, DebertaConfig, DebertaV2Config, DistilBertConfig, DPRConfig, ElectraConfig, ErnieConfig, ErnieMConfig, EsmConfig, FalconConfig, FastSpeech2ConformerConfig, FlaubertConfig, FNetConfig, FSMTConfig, FunnelConfig, GemmaConfig, Gemma2Config, GitConfig, GPT2Config, GPT2Config, GPTBigCodeConfig, GPTNeoConfig, GPTNeoXConfig, GPTNeoXJapaneseConfig, GPTJConfig, GPTSanJapaneseConfig, GroundingDinoConfig, GroupViTConfig, HubertConfig, IBertConfig, IdeficsConfig, Idefics2Config, InstructBlipConfig, InstructBlipVideoConfig, JambaConfig, JetMoeConfig, JukeboxConfig, Kosmos2Config, LayoutLMConfig, LayoutLMv2Config, LayoutLMv3Config, LEDConfig, LiltConfig, LlamaConfig, LlavaConfig, LlavaNextVideoConfig, LlavaNextConfig, LongformerConfig, LongT5Config, LukeConfig, LxmertConfig, M2M100Config, MambaConfig, MarianConfig, MBartConfig, MegaConfig, MegatronBertConfig, MgpstrConfig, MistralConfig, MixtralConfig, MobileBertConfig, MPNetConfig, MptConfig, MraConfig, MT5Config, MusicgenConfig, MusicgenMelodyConfig, MvpConfig, NezhaConfig, NllbMoeConfig, NystromformerConfig, OlmoConfig, OneFormerConfig, OpenAIGPTConfig, OPTConfig, Owlv2Config, OwlViTConfig, PaliGemmaConfig, PegasusConfig, PegasusXConfig, PerceiverConfig, PersimmonConfig, PhiConfig, Phi3Config, Pix2StructConfig, PLBartConfig, ProphetNetConfig, QDQBertConfig, Qwen2Config, Qwen2MoeConfig, RagConfig, RealmConfig, RecurrentGemmaConfig, ReformerConfig, RemBertConfig, RetriBertConfig, RobertaConfig, RobertaPreLayerNormConfig, RoCBertConfig, RoFormerConfig, RwkvConfig, SeamlessM4TConfig, SeamlessM4Tv2Config, SiglipConfig, Speech2TextConfig, Speech2Text2Config, SpeechT5Config, SplinterConfig, SqueezeBertConfig, StableLmConfig, Starcoder2Config, SwitchTransformersConfig, T5Config, TapasConfig, TransfoXLConfig, TvpConfig, UdopConfig, UMT5Config, VideoLlavaConfig, ViltConfig, VipLlavaConfig, VisualBertConfig, VitsConfig, Wav2Vec2Config, Wav2Vec2BertConfig, Wav2Vec2ConformerConfig, WhisperConfig, XCLIPConfig, XGLMConfig, XLMConfig, XLMProphetNetConfig, XLMRobertaConfig, XLMRobertaXLConfig, XLNetConfig, XmodConfig, YosoConfig.

In [ ]:
!pip install --quiet transformers

from transformers import AutoTokenizer, AutoModelForSequenceClassification, AutoConfig
import torch

model_name = "vectara/hallucination_evaluation_model"

# Explicitly load the configuration
config = AutoConfig.from_pretrained(model_name, trust_remote_code=True)

# Pass the configuration to the tokenizer
tokenizer = AutoTokenizer.from_pretrained(model_name, config=config, trust_remote_code=True)

model = AutoModelForSequenceClassification.from_pretrained(model_name, trust_remote_code=True)

You are using a model of type HHEMv2Config to instantiate a model of type HHEMv2. This is not supported for all configurations of models and can yield errors.


ValueError: Unrecognized configuration class <class 'transformers_modules.vectara.hallucination_evaluation_model.2c44150de684dbcbc6eb2abd9e92b428c5fb3361.configuration_hhem_v2.HHEMv2Config'> to build an AutoTokenizer.
Model type should be one of AlbertConfig, AlignConfig, BarkConfig, BartConfig, BertConfig, BertGenerationConfig, BigBirdConfig, BigBirdPegasusConfig, BioGptConfig, BlenderbotConfig, BlenderbotSmallConfig, BlipConfig, Blip2Config, BloomConfig, BridgeTowerConfig, BrosConfig, CamembertConfig, CanineConfig, ChineseCLIPConfig, ClapConfig, CLIPConfig, CLIPSegConfig, ClvpConfig, LlamaConfig, CodeGenConfig, CohereConfig, ConvBertConfig, CpmAntConfig, CTRLConfig, Data2VecAudioConfig, Data2VecTextConfig, DbrxConfig, DebertaConfig, DebertaV2Config, DistilBertConfig, DPRConfig, ElectraConfig, ErnieConfig, ErnieMConfig, EsmConfig, FalconConfig, FastSpeech2ConformerConfig, FlaubertConfig, FNetConfig, FSMTConfig, FunnelConfig, GemmaConfig, Gemma2Config, GitConfig, GPT2Config, GPT2Config, GPTBigCodeConfig, GPTNeoConfig, GPTNeoXConfig, GPTNeoXJapaneseConfig, GPTJConfig, GPTSanJapaneseConfig, GroundingDinoConfig, GroupViTConfig, HubertConfig, IBertConfig, IdeficsConfig, Idefics2Config, InstructBlipConfig, InstructBlipVideoConfig, JambaConfig, JetMoeConfig, JukeboxConfig, Kosmos2Config, LayoutLMConfig, LayoutLMv2Config, LayoutLMv3Config, LEDConfig, LiltConfig, LlamaConfig, LlavaConfig, LlavaNextVideoConfig, LlavaNextConfig, LongformerConfig, LongT5Config, LukeConfig, LxmertConfig, M2M100Config, MambaConfig, MarianConfig, MBartConfig, MegaConfig, MegatronBertConfig, MgpstrConfig, MistralConfig, MixtralConfig, MobileBertConfig, MPNetConfig, MptConfig, MraConfig, MT5Config, MusicgenConfig, MusicgenMelodyConfig, MvpConfig, NezhaConfig, NllbMoeConfig, NystromformerConfig, OlmoConfig, OneFormerConfig, OpenAIGPTConfig, OPTConfig, Owlv2Config, OwlViTConfig, PaliGemmaConfig, PegasusConfig, PegasusXConfig, PerceiverConfig, PersimmonConfig, PhiConfig, Phi3Config, Pix2StructConfig, PLBartConfig, ProphetNetConfig, QDQBertConfig, Qwen2Config, Qwen2MoeConfig, RagConfig, RealmConfig, RecurrentGemmaConfig, ReformerConfig, RemBertConfig, RetriBertConfig, RobertaConfig, RobertaPreLayerNormConfig, RoCBertConfig, RoFormerConfig, RwkvConfig, SeamlessM4TConfig, SeamlessM4Tv2Config, SiglipConfig, Speech2TextConfig, Speech2Text2Config, SpeechT5Config, SplinterConfig, SqueezeBertConfig, StableLmConfig, Starcoder2Config, SwitchTransformersConfig, T5Config, TapasConfig, TransfoXLConfig, TvpConfig, UdopConfig, UMT5Config, VideoLlavaConfig, ViltConfig, VipLlavaConfig, VisualBertConfig, VitsConfig, Wav2Vec2Config, Wav2Vec2BertConfig, Wav2Vec2ConformerConfig, WhisperConfig, XCLIPConfig, XGLMConfig, XLMConfig, XLMProphetNetConfig, XLMRobertaConfig, XLMRobertaXLConfig, XLNetConfig, XmodConfig, YosoConfig.

In [ ]:
from transformers import pipeline, AutoTokenizer

In [ ]:
classifier = pipeline(
            "text-classification",
            model='vectara/hallucination_evaluation_model',
            tokenizer=AutoTokenizer.from_pretrained('google/flan-t5-base'),
            trust_remote_code=True,
            padding = True,
            truncation = True,

        )

You are using a model of type HHEMv2Config to instantiate a model of type HHEMv2. This is not supported for all configurations of models and can yield errors.
You are using a model of type HHEMv2Config to instantiate a model of type HHEMv2. This is not supported for all configurations of models and can yield errors.


In [ ]:
scores = classifier.predict([
    ("A man walks into a bar and buys a drink", " A bloke swigs alcohol at a pub")
])

In [ ]:
scores

[{'label': 'hallucinated', 'score': 0.5528202056884766}]

In [ ]:
score = classifier.predict([
    ("A man walks into a bar and buys a drink", "A bloke swigs alcohol at a pub"),
    ("A person who drivs once in a week can be treated as a senior age person","A senior age person who goes grocery for shopping once in a week"),
    ("I like to eat apple as it is my favourite fruit","I don't like to eat apple as this is not my favourite fruit")
  ])


In [ ]:
score

[{'label': 'hallucinated', 'score': 0.5528202056884766},
 {'label': 'hallucinated', 'score': 0.8735398054122925},
 {'label': 'consistent', 'score': 0.8304786086082458}]

In the context of your example, the classifier is using a model to evaluate the relationship between pairs of sentences, with outputs indicating whether the sentences are "hallucinated" or "consistent." Here’s a breakdown of the results:

## Understanding the Labels
### Hallucinated:
This label suggests that the second sentence does not align well with the context or meaning of the first sentence. The model might determine that the second sentence introduces information or details that are inconsistent or not directly supported by the first sentence.

### Consistent:
This label indicates that the second sentence aligns well with the context or meaning of the first sentence. The information in the second sentence is consistent with what is presented in the first sentence.

## Explanation of Results
### Pair 1:

Sentences:
"A man walks into a bar and buys a drink"
"A bloke swigs alcohol at a pub"
Score: 'label': 'hallucinated', 'score': 0.5528202056884766
Explanation: The sentences are similar in meaning, but the model labeled this pair as "hallucinated." This could be due to subtle differences in context or details that make the second sentence not perfectly consistent with the first one. For example, "a bloke" vs. "a man" or "swigs alcohol" vs. "buys a drink" might introduce slight discrepancies in meaning or context.

### Pair 2:

Sentences:
"A person who drives once in a week can be treated as a senior age person"
"A senior age person who goes grocery shopping once in a week"
Score: 'label': 'hallucinated', 'score': 0.8735398054122925
Explanation: This pair is also labeled as "hallucinated" with a high score. The first sentence suggests a generalization about senior age people based on their driving frequency, while the second sentence specifically mentions grocery shopping. The classifier may have determined that the connection between driving frequency and senior age is not well-supported or directly reflected in the second sentence.

### Pair 3:

Sentences:
"I like to eat apple as it is my favourite fruit"
"I don't like to eat apple as this is not my favourite fruit"
Score: 'label': 'consistent', 'score': 0.8304786086082458
Explanation: This pair is labeled as "consistent," which means that despite the sentences expressing opposing preferences, they are consistent in their internal logic. Each sentence reflects a personal preference about apples and is logically consistent with itself. The classifier recognizes that the two sentences are expressing opposing but valid personal opinions.

### Summary

Hallucinated pairs are where the second sentence introduces information or nuances that the model considers inconsistent with the first sentence.
Consistent pairs align well with the context or meaning of the first sentence, even if they express opposing views.
The scores represent the confidence level of the model's classification. Higher scores indicate stronger confidence in the label assigned.